src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,68 @@
+package duke.task;
+
+/**
+ * Class to represent Task (not to be instantiated, hence abstract).
+ *
+ * @author Melissa Anastasia Harijanto
+ */
+public abstract class Task {
+    /** String representation of the task name. */
+    protected String taskName;
+
+    /** Boolean value to decide whether a task is done. */
+    protected boolean isDone;
+
+    /**
+     * Constructor for the Task class.
+     *
+     * @param taskName The name of the task.
+     */
+    protected Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of a task; if done, 'X' will
+     * be returned; otherwise a space will be returned.
+     *
+     * @return Status icon of a task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks a task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks a task as undone.
+     */
+    public void unmarkAsDone() {
```

Comment:
hmm, I would consider just using mark and unmark as the method names? unmark as done might be slightly confusing?
but again, really digging around to comment on here, so no worries.